Tests/AsyncConsoleKitTests/AsyncUtilities.swift

```diff
@@ -0,0 +1,127 @@
+import ConsoleKit
+import XCTest
+
+#if swift(>=5.5)
+extension String: Error {}
+
+final class TestGroup: AsyncCommandGroup {
+    struct Signature: CommandSignature {
+        @Flag(name: "version", help: "Prints the version")
+        var version: Bool
+        init() { }
+    }
+
+    let commands: [String : AnyAsyncCommand] = [
+        "test": TestCommand(),
+        "sub": SubGroup()
+    ]
+
+    let help: String = "This is a test grouping!"
+
+    func run(using context: CommandContext, signature: Signature) throws {
+        if signature.version {
+            context.console.print("v2.0")
+        }
+    }
+}
+
+final class SubGroup: AsyncCommandGroup {
+    struct Signature: CommandSignature {
+        @Flag(name: "version", help: "Prints the version")
+        var version: Bool
+        init() { }
+    }
+
+    let commands: [String : AnyAsyncCommand] = [
+        "test": TestCommand()
+    ]
+
+    let help: String = "This is a test sub grouping!"
+
+    func run(using context: CommandContext, signature: Signature) throws {
+        if signature.version {
+            context.console.print("v2.0")
+        }
+    }
+}
+
+final class TestCommand: AsyncCommand {
+    struct Signature: CommandSignature {
+        @Argument(name: "foo", help: """
+        A foo is required
+        An error will occur if none exists
+        """)
+        var foo: String
+
+        @Option(name: "bar", short: "b", help: """
+        Add a bar if you so desire
+        Try passing it
+        """)
+        var bar: String?
+
+        @Flag(name: "baz", short: "B", help: """
+        Add a baz if you so desire
+        It's just a flag
+        """)
+        var baz: Bool
+
+        init() { }
+    }
+
+    let help: String = "This is a test command"
+
+    func run(using context: CommandContext, signature: Signature) throws {
+        XCTAssertEqual(signature.$foo.name, "foo")
+        XCTAssertEqual(signature.$bar.name, "bar")
+        XCTAssertEqual(signature.$baz.name, "baz")
+        context.console.output("Foo: \(signature.foo) Bar: \(signature.bar ?? "nil")".consoleText(.info))
+    }
+}
+
+final class StrictCommand: AsyncCommand {
+    struct Signature: CommandSignature {
+        @Argument(name: "number")
+        var number: Int
+        
+        @Argument(name: "bool")
+        var bool: Bool
+        
+        init() { }
+    }
+    var help: String = "I error if you pass in bad values"
+
+    func run(using context: CommandContext, signature: Signature) throws {
```

Comment:
This should be async. Can you create a new function that's async that we run from here. It doesn't need to do anything async, we just need to make sure suspending in a command works ok